File: 53-maximum-subarray/53-maximum-subarray.java
class Solution {
    public int max(int a, int b){
        if(a > b) return a;
        return b;
    }
    public int maxSubArray(int[] nums) {
        int n = nums.length;
        int maxSum = nums[0];
        int currSum= nums[0];
        
        for(int i=1; i<n; i++){
            currSum = max(currSum+nums[i], nums[i]);
            maxSum = max(currSum, maxSum);
        }
        
        return maxSum;
    }
}

File: 53-maximum-subarray/53-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        n = len(nums)
        maxSum = nums[0]
        currSum = nums[0]
        
        for i in range(1, n) : 
            currSum = max(currSum+nums[i], nums[i])
            maxSum = max(maxSum, currSum)
            
        return maxSum
        